source/heating_model/shear_heating.cc

```diff
@@ -41,15 +41,30 @@ namespace aspect
 
       for (unsigned int q=0; q<heating_model_outputs.heating_source_terms.size(); ++q)
         {
-          const SymmetricTensor<2,dim> compressible_strain_rate
-            = (this->get_material_model().is_compressible()
-               ?
-               material_model_inputs.strain_rate[q] - 1./3. * trace(material_model_inputs.strain_rate[q]) * unit_symmetric_tensor<dim>()
-               :
-               material_model_inputs.strain_rate[q]);
-
-          heating_model_outputs.heating_source_terms[q] = 2.0 * material_model_outputs.viscosities[q] *
-                                                          compressible_strain_rate * compressible_strain_rate;
+          const SymmetricTensor<4,dim> C = material_model_outputs.constitutive_tensors[q];
+          const SymmetricTensor<2,dim> directed_strain_rate = ((C == dealii::identity_tensor<dim> ())
```

Comment:
I agree, this is awkward to read. And maybe it's some compiler magic, but doing the comparison seems to make no difference in assembly time, and doing the multiplication slows down the entire model by an order of magnitude. Since I'm just using the identity tensor as a sort of "signaling" null value, is there a more obvious object to use in its place? If it doesn't make a difference, I kinda like using the identity tensor because it's sort of self-documenting in a way.